src/app/public/modules/file-attachment/file-attachment.component.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  ViewChild
+} from '@angular/core';
+
+import {
+  SkyFileItem
+} from './file-item';
+
+import {
+  SkyFileAttachmentChange
+} from './file-attachment-change';
+
+import {
+  SkyFileAttachmentService
+} from './file-attachment.service';
+
+import {
+  SkyFileItemService
+} from './file-item.service';
+
+@Component({
+  selector: 'sky-file-attachment',
+  templateUrl: './file-attachment.component.html',
+  styleUrls: ['./file-attachment.component.scss'],
+  providers: [
+    SkyFileAttachmentService,
+    SkyFileItemService
+  ]
+})
+export class SkyFileAttachmentComponent implements OnDestroy {
+  @Output()
+  public fileChange = new EventEmitter<SkyFileAttachmentChange>();
+
+  @Input()
+  public minFileSize: number = 0;
+
+  @Input()
+  public maxFileSize: number = 500000;
+
+  @Input()
+  public validateFn: Function;
+
+  @Input()
+  public acceptedTypes: string;
+
+  @Input()
+  public required: boolean = false;
+
+  @ViewChild('fileInput')
+  private inputEl: ElementRef;
+
+  @ViewChild('labelWrapper')
+  private labelWrap: ElementRef;
+
+  public rejectedOver: boolean = false;
+  public acceptedOver: boolean = false;
+  public singleFileAttachment: SkyFileItem;
+
+  private enterEventTarget: any;
```

Comment:
Code style > Check your ordering of properties here: https://host.nxt.blackbaud.com/skyux-team-docs/style#class-properties